File: getToken.js
import dotenv from 'dotenv'
import jwt from 'jsonwebtoken'
import { v4 as uuid } from 'uuid'
import fetch from 'node-fetch';
dotenv.config()

function generateJsonWebToken() {
    const issueAt = Math.round(new Date().getTime() / 1000);
    const expirationAt = issueAt + 300;

    const payload = {
        "aud": process.env.TOKEN_PROD_URL + "?realm=aolcorporate/aolexternals",
        "iat": issueAt,
        "exp": expirationAt,
        "sub": process.env.CLIENT_ID,
        "iss": process.env.CLIENT_ID,
        "jti": uuid()
    };

    const token = jwt.sign(payload, process.env.CLIENT_SECRET, { algorithm: 'HS256' });
    return token;
}

async function getOauthToken() {
    const headers = {
        "Content-Type": "application/x-www-form-urlencoded",
        "Accept": "application/json"
    };
    const payload = {
        "grant_type": "client_credentials",
        "scope": "one",
        "realm": "aolcorporate/aolexternals",
        "client_assertion_type": "urn:ietf:params:oauth:client-assertion-type:jwt-bearer",
        "client_assertion": generateJsonWebToken()
    };
    const request = {
        "method": "post",
        "headers": headers,
        "body": new URLSearchParams(payload)
    };

    const tokenRequest = await fetch(process.env.TOKEN_PROD_URL, request)
    const token = await tokenRequest.json()

    return token
}

const token = await getOauthToken();
console.log(token)

File: getToken.py
import json
import time
import uuid
import requests
import jwt

token_stage_url = 'https://id-uat.b2b.verizonmedia.com/identity/oauth2/access_token'
token_prod_url = 'https://id.b2b.verizonmedia.com/identity/oauth2/access_token'
token_url = token_prod_url
client_id = '8a28g22c-d2e1-4553-9998-b03bb11ddd4g'
secret = 'rAN0vhFKKRNyLbdOQ3vuVU3xI4+dI6RCoWrLiypiiBZJc+BoOp'

def get_jwt_token(client_id, secret):
    issueAt = int(time.time())
    expirationAt = int(time.time()) + 300 # 5 minutes

    payload = {
        'aud': token_url + "?realm=dsp",
        'iss': client_id,
        'sub': client_id,
        'exp': expirationAt,
        'iat': issueAt,
        'jti': str(uuid.uuid4())
    }

    header = {
        "alg": "HS256",
        "typ": "JWT"
    }

    encoded = jwt.encode(payload, secret, algorithm = 'HS256', headers = header)
    return encoded


def get_oauth_token():
    token = get_jwt_token('8a28g22c-d2e1-4553-9998-b03bb11ddd4g', 'rAN0vhFKKRNyLbdOQ3vuVU3xI4+dI6RCoWrLiypiiBZJc+BoOp')

    payload = "grant_type=client_credentials" \
              "&scope=dsp-api-access" \
              "&realm=dsp" \
              "&client_assertion_type=urn:ietf:params:oauth:client-assertion-type:jwt-bearer" \
              "&client_assertion=" + token

    headers = {
        'Content-Type': "application/x-www-form-urlencoded",
        'Accept': "application/json"
    }

    response = requests.post(token_url, data = payload, headers = headers)
    print('-----------START-----------')
    print('POST',response.request.url)
    print('\r\n')
    print(response.request.body)
    print('\r\n')
    print(response.request.headers)
    print('-----------END--------------')

    return response.text

print('Response =>', json.loads(get_oauth_token()))
